scripts/screeps_rl_experiment_card.py

```
@@ -1232,26 +1354,65 @@ def main(
+        source_gate = None
+        if args.from_latest_accepted_dataset and dataset_run_id is not None:
+            raise CardValidationError("--from-latest-accepted-dataset cannot be combined with --dataset-run-id")
+        if args.source_gate_id is not None:
+            source_gate = select_accepted_dataset_gate(args.dataset_gate_root, args.source_gate_id)
+            source_dataset_run_id = str(source_gate["dataset_run_id"])
+            if dataset_run_id is not None and dataset_run_id != source_dataset_run_id:
+                raise CardValidationError("--dataset-run-id must match --source-gate-id dataset_run_id")
+            dataset_run_id = source_dataset_run_id
+        elif args.from_latest_accepted_dataset or args.loop_a_local_fallback:
             if dataset_run_id is not None:
                 raise CardValidationError("--from-latest-accepted-dataset cannot be combined with --dataset-run-id")
```

Ruff (0.15.12)

[warning] Abstract `raise` to an inner function

(TRY301)